client/src/main/java/net/spals/oembed4j/client/JerseyOEmbedClient.java

```diff
@@ -0,0 +1,140 @@
+import java.io.IOException;
+import java.io.InputStream;
+import java.security.SecureRandom;
+import java.security.cert.X509Certificate;
+import java.util.Optional;
+
+/**
+ * An implementation of {@link OEmbedClient} based
+ * on a Jersey client.
+ *
+ * @author tkral
+ */
+public final class JerseyOEmbedClient implements OEmbedClient {
+
+    private static final X509TrustManager DEFAULT_TRUST_MANAGER = new X509TrustManager() {
+        @Override
+        public X509Certificate[] getAcceptedIssuers() {
+            return new X509Certificate[0];
+        }
+
+        @Override
+        public void checkClientTrusted(final X509Certificate[] x509Certificates, final String s) {  }
+
+        @Override
+        public void checkServerTrusted(final X509Certificate[] x509Certificates, final String s) {  }
+    };
+
+    public static JerseyOEmbedClient create(final OEmbedRegistry registry) {
+        try {
+            // Configure the Jersey client
+            // Setup SSL management
+            // TODO: DO we need some real SSL management?
+            final SSLContext sslContext = SSLContext.getInstance("TLS");
+            sslContext.init(null /*KeyManagers*/, new TrustManager[] {DEFAULT_TRUST_MANAGER}, new SecureRandom());
+
+            final ClientBuilder clientBuilder = ClientBuilder.newBuilder();
+            clientBuilder.sslContext(sslContext).hostnameVerifier((s, sslSession) -> true);
+            clientBuilder.property(ClientProperties.FOLLOW_REDIRECTS, Boolean.TRUE);
+
+            return new JerseyOEmbedClient(clientBuilder.build(), registry);
+        } catch (Exception e) {
+            throw Throwables.propagate(e);
+        }
+    }
+
+    private final Client client;
+    private final OEmbedRegistry registry;
+    private final OEmbedResponseCache responseCache;
+    private final OEmbedResponseParser responseParser;
+
+    @VisibleForTesting
+    JerseyOEmbedClient(final Client client,
+                       final OEmbedRegistry registry) {
+        this.client = client;
+        this.registry = registry;
+        this.responseCache = OEmbedResponseCache.create(this);
+        this.responseParser = new OEmbedResponseParser();
+    }
+
+    @VisibleForTesting
+    JerseyOEmbedClient(final Client client,
+                       final OEmbedRegistry registry,
+                       final OEmbedResponseCache responseCache,
+                       final OEmbedResponseParser responseParser) {
+        this.client = client;
+        this.registry = registry;
+        this.responseCache = responseCache;
+        this.responseParser = responseParser;
+    }
+
+    /**
+     * @see OEmbedClient#close()
+     */
+    @Override
+    public void close() {
+        client.close();
+    }
+
+    /**
+     * @see OEmbedClient#execute(OEmbedRequest)
+     */
+    @Override
+    public Optional<OEmbedResponse> execute(final OEmbedRequest request) {
+        return responseCache.get(request.getResourceURI());
```

Comment:
There's also a caching element to the oEmbed spec which I tried to capture. But I think it's slightly broken + it needs testing so that's probably something we can do on Sunday.